DQM/SiPixelHeterogeneous/plugins/SiPixelCompareTrackSoAAlpaka.cc

```diff
@@ -0,0 +1,318 @@
+      float dr2 = reco::deltaR2(etaCPU, phiCPU, etaGPU, phiGPU);
+      if (dr2 > dr2cut_)
+        continue;  // this is arbitrary
+      if (mindr2 > dr2) {
+        mindr2 = dr2;
+        closestTkidx = gid;
+      }
+    }
+
+    hpt_eta_tkAllCPU_->Fill(etaCPU, ptCPU);  //all CPU tk
+    hphi_z_tkAllCPU_->Fill(phiCPU, zipCPU);
+    if (closestTkidx == notFound)
+      continue;
+    nLooseAndAboveTracksCPU_matchedGPU++;
+
+    hchi2_->Fill(tsoaCPU.view()[it].chi2(), tsoaGPU.view()[closestTkidx].chi2());
+    hCharge_->Fill(helper::charge(tsoaCPU.view(), it), helper::charge(tsoaGPU.view(), closestTkidx));
+    hnHits_->Fill(helper::nHits(tsoaCPU.view(), it), helper::nHits(tsoaGPU.view(), closestTkidx));
+    hnLayers_->Fill(tsoaCPU.view()[it].nLayers(), tsoaGPU.view()[closestTkidx].nLayers());
+    hpt_->Fill(tsoaCPU.view()[it].pt(), tsoaGPU.view()[closestTkidx].pt());
+    hptLogLog_->Fill(tsoaCPU.view()[it].pt(), tsoaGPU.view()[closestTkidx].pt());
+    heta_->Fill(etaCPU, tsoaGPU.view()[closestTkidx].eta());
+    hphi_->Fill(phiCPU, helper::phi(tsoaGPU.view(), closestTkidx));
+    hz_->Fill(zipCPU, helper::zip(tsoaGPU.view(), closestTkidx));
+    htip_->Fill(tipCPU, helper::tip(tsoaGPU.view(), closestTkidx));
+    hptdiffMatched_->Fill(ptCPU - tsoaGPU.view()[closestTkidx].pt());
+    hCurvdiffMatched_->Fill((helper::charge(tsoaCPU.view(), it) / tsoaCPU.view()[it].pt()) -
+                            (helper::charge(tsoaGPU.view(), closestTkidx) / tsoaGPU.view()[closestTkidx].pt()));
+    hetadiffMatched_->Fill(etaCPU - tsoaGPU.view()[closestTkidx].eta());
+    hphidiffMatched_->Fill(reco::deltaPhi(phiCPU, helper::phi(tsoaGPU.view(), closestTkidx)));
+    hzdiffMatched_->Fill(zipCPU - helper::zip(tsoaGPU.view(), closestTkidx));
+    htipdiffMatched_->Fill(tipCPU - helper::tip(tsoaGPU.view(), closestTkidx));
+    hpt_eta_tkAllCPUMatched_->Fill(etaCPU, tsoaCPU.view()[it].pt());  //matched to gpu
+    hphi_z_tkAllCPUMatched_->Fill(etaCPU, zipCPU);
+  }
+  hnTracks_->Fill(nTracksCPU, nTracksGPU);
+  hnLooseAndAboveTracks_->Fill(nLooseAndAboveTracksCPU, nLooseAndAboveTracksGPU);
+  hnLooseAndAboveTracks_matched_->Fill(nLooseAndAboveTracksCPU, nLooseAndAboveTracksCPU_matchedGPU);
+}
+
+//
+// -- Book Histograms
+//
+template <typename T>
+void SiPixelCompareTrackSoAAlpaka<T>::bookHistograms(DQMStore::IBooker& iBook,
+                                                     edm::Run const& iRun,
+                                                     edm::EventSetup const& iSetup) {
+  iBook.cd();
+  iBook.setCurrentFolder(topFolderName_);
+
+  // clang-format off
+  std::string toRep = "Number of tracks";
+  // FIXME: all the 2D correlation plots are quite heavy in terms of memory consumption, so a as soon as DQM supports THnSparse
+  // these should be moved to a less resource consuming format
+  hnTracks_ = iBook.book2I("nTracks", fmt::sprintf("%s per event; CPU; GPU",toRep), 501, -0.5, 500.5, 501, -0.5, 500.5);
```

Comment:
In all cases of `fmt::sprintf()` the `fmt::format()` would be as easy to use, and it would be tiny bit easier to migrate to `std::format()` once we get there
```suggestion
  hnTracks_ = iBook.book2I("nTracks", fmt::format("{} per event; CPU; GPU",toRep), 501, -0.5, 500.5, 501, -0.5, 500.5);
```